### ISSUE TITLE
Feature request: prevent redirect to vendor directory

### ISSUE THREAD

**Issue by DanielRuf** [Codeception/Codeception#5495]
#### What are you trying to achieve?
As a developer / user I want to create a `tools` directory on the same level as `composer.json` to use different versions of Codeception.

#### What do you get instead?

https://github.com/Codeception/Codeception/blob/60702493e37019ca37009b229ed055b30d62b466/autoload.php#L4

```bash
==== Redirecting to Composer-installed version in vendor/codeception ====
```

### What do I expect
A CLI parameter to prevent this redirect.

### Details

* Codeception version: 3.0.0
* PHP Version: 7.2
* Operating System: macOS
* Installation type: Phar
* List of installed packages (`composer show`)
* Suite configuration:

**Comment by Naktibalda** [comment]
How about not installing Codeception to vendor dir if you don't want it loaded?

**Comment by DanielRuf** [comment]
> How about not installing Codeception to vendor dir if you don't want it loaded?

This is not possible in our case as we have a TYPO3 project and it installs `codeception/codeception` for its internal testsuites.

**Comment by DanielRuf** [comment]
And this might be the same case in other projects too and affects others too.

**Comment by Naktibalda** [comment]
This redirect is here for a reason - it prevents odd errors when different versions of Codeception or dependencies are installed in vendor dir and external locations and some classes are loaded from one place and some - from another.

But you are welcome to implement this parameter in your fork and test it in your project.
If it works, please make a pull request.

**Comment by DanielRuf** [comment]
Well the phar is bundled and separated from `vendor` in general so there should be no conflicts in general afaik ;-)

I am aware of the fact that it makes sense for globally composer-installed + locally composer-installed versions (but not for the phar bundles).

**Comment by DanielRuf** [comment]
> But you are welcome to implement this parameter in your fork and test it in your project.
> If it works, please make a pull request.

Will do so.

**Comment by Naktibalda** [comment]
> Well the phar is bundled and separated from `vendor` in general so there should be no conflicts in general afaik ;-)

Conflicts start happenning when Codeception includes `vendor/autoload/autoload.php` which is necessary for functional and unit tests.

Here is a couple of examples: https://github.com/Codeception/Codeception/issues/3999
https://github.com/Codeception/Codeception/issues/3999

**Comment by DanielRuf** [comment]
> Conflicts start happenning when Codeception includes `vendor/autoload/autoload.php` which is necessary for functional and unit tests.

We currently use the acceptance tests and the redirect prevents that we can use a separate version.
Also phar bundles should be standalone and if we need other phars for functional or unit tests (eg phpunit) we can probably add them or let Codeception reuse them.

The only issue at the moment is the initial redirect.

**Comment by DanielRuf** [comment]
Actually the examples do not cover our usecase ;-)

**Comment by Naktibalda** [comment]
I got 2 alternatives to you:
1. If you can't remove `codeception/codeception` from vendor directory, then remove tests from project directory. There is no reason for acceptance tests to be in the same codebase with project files.
2. Use your own copy of [codecept](https://github.com/Codeception/Codeception/blob/3.0/codecept) and [autoload.php](https://github.com/Codeception/Codeception/blob/3.0/autoload.php) to bypass redirect.

**Comment by DanielRuf** [comment]
1 and 2 are no solutions for us.

1 not because we ship our own testsuites with complete TYPO3 projects and 2 because it makes integration too complicated and we directly use the provided builds.

Do not overcomplicate things.

**Comment by DanielRuf** [comment]
Also your proposed solution 1 would not work when yu have any composer dependency as this would create `autoload.php` in `vendor`.

https://github.com/Codeception/Codeception/blob/3.0/autoload.php#L4

Even phpunit can be used like this (use separate phar instead of the composer installed version) afaik.

**Comment by Naktibalda** [comment]
Implemented in #5498 and released in 3.0.2
